src/WebJobs.Extensions.DurableTask/DurableHttpRequest.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Runtime.Serialization;
+using Microsoft.Extensions.Primitives;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    /// <summary>
+    /// Request used to make an HTTP call through Durable Functions.
+    /// </summary>
+    public class DurableHttpRequest
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DurableHttpRequest"/> class.
+        /// </summary>
+        /// <param name="method">Method used for HTTP request.</param>
+        /// <param name="uri">Uri used to make the HTTP request.</param>
+        /// <param name="headers">Headers added to the HTTP request.</param>
+        /// <param name="content">Content added to the body of the HTTP request.</param>
+        /// <param name="tokenSource">AAD authentication attached to the HTTP request.</param>
+        /// <param name="asynchronousPatternEnabled">Specifies whether the DurableHttpRequest should handle the asynchronous pattern.</param>
+        public DurableHttpRequest(
+            HttpMethod method,
+            Uri uri,
+            IDictionary<string, StringValues> headers = null,
+            string content = null,
+            ITokenSource tokenSource = null,
+            bool asynchronousPatternEnabled = true)
+        {
+            this.Method = method;
+            this.Uri = uri;
+            this.Headers = headers ?? new Dictionary<string, StringValues>(StringComparer.OrdinalIgnoreCase);
```

Comment:
It's fine to take the headers as a parameter, but it's dangerous to assign it directly to your `Headers` property because the caller might not have created their dictionary with `StringComparer.OrdinalIgnoreCase`. To avoid problems related to this, it is better to copy the values from the parameter dictionary into a dictionary object you created yourself.